Tidy up navbar font; minor variable name changes.

seanlip opened the pull request.

seanlip commented:
Hi @giritheja, mind taking a review pass at this (both in terms of code + UI)? We're planning to do a release soon, so I thought I'd tidy up the look and feel of the navbar a bit before that.

Thanks!

seanlip commented:
PS: to check out the branch on your local machine, do the following:

```
    git pull
    git checkout navbar-font
```

giritheja commented:
Hey, @seanlip  Yeah sure. I will take a look.

giritheja commented:
@seanlip The code looks good. Here are some things you may want to take a look at:
![screenshot from 2016-04-01 13 09 32](https://cloud.githubusercontent.com/assets/8845039/14201787/b1ba58b8-f811-11e5-9e31-95afc321cba5.png)
1. The dropdown alignment with the about tab and profile tab bothers me when I look at it. 
![screenshot from 2016-04-01 13 36 03](https://cloud.githubusercontent.com/assets/8845039/14201850/236d1c2a-f812-11e5-9dd5-fe7bc0a749e8.png)
2. When a user resizes the window to a small size and clicks the Hamburger menu, the side menu appears. If the user maximizes back to the full-screen size then the side menu stays. The user has to click somewhere in the page to make it disappear.
3. I think the sign in dropdown does not add up to the navbar design.
4. Can all the tabs in the navbar look the same?
    Sign in and Create buttons can be shown when the window is resized to smaller size.
5. If a user(signed in) resizes the window to a smaller size then profile tab is shown in the navbar but if the user is not signed in then there is no sign in button shown in the navbar.

seanlip commented:
Hi @giritheja, thank you very much for the review! In response to your points:

1. This is a great catch -- I missed it. Thank you for spotting it! It's now fixed.
2. This is also a good catch. It's now fixed.
3, 4. The "Create" and "Sign in" buttons are intentional; these two things perform actions, and are not just for navigation, so it makes sense to distinguish them from the others. However, I agree that the sign-in dropdown looks a bit awkward. For the purposes of this commit, though, I'm planning to leave this as-is, and file an issue once this PR goes in.
5. This is a good catch, too! It's now fixed.

I've pushed another commit; please could you take a look and let me know what you think? Thanks!

giritheja commented:
Hey, @seanlip . All the previous comments are fixed. Here are some little things you may want to take a look at:
1. When a user clicks blog link in either sidemenu or in the navbar, It redirects to "site/blog/url". If a user clicks blog in the navbar, the link opens in the current window. If a user clicks blog in sidemenu, It opens in a new window. 
2.This is a minor issue. If a user hovers mouse over create or login button, this happens
![screenshot from 2016-04-02 14 22 39](https://cloud.githubusercontent.com/assets/8845039/14225458/1fde4910-f8e1-11e5-94bf-c5e06745d025.png)
The browser suggests that you will go to the page in which you are when you click it (It happens in all pages). This behaviour is not observed when a user is logged in and is on the gallery page/ forum page.

seanlip commented:
Thanks! Addressed both of these; PTAL. (Note that the blog link is specified in the admin page, so site/blog/url is expected. I made the navbar open it in a new tab.)

giritheja commented on core/templates/dev/head/base.html:
```diff
@@ -212,11 +212,14 @@
+                            </div>
+                            <ul class="dropdown-menu oppia-navbar-dropdown" role="menu" style="margin-right: 15px; padding: 0;" ng-mouseover="onMouseoverDropdownMenu($event)" ng-mouseleave="onMouseoutDropdownMenu($event)">
                               <li>
                                 <a href style="padding: 0; width: 200px;" ng-click="onLoginButtonClicked('{{login_url}}')">
```
Is href required here?

seanlip replied:
I'm leaving it in; I think it's needed for accessibility.

giritheja commented:
@seanlip Sorry, I should have mentioned the above in my previous comment. One last thing, If a user is logged in and resizes the window to a smaller size then the user is not provided option to create explorations but the user has the ability to edit his/her explorations (which are already created). If it is meant to be like that then everything LGTM.

seanlip commented:
I thought about it, and it seems fine to have the "Create" button display on smaller widths than it currently does, so I changed the cutoff. Thanks for the suggestion!

PTAL, and if you like it, I'll merge this. Thanks!

giritheja commented:
Hey @seanlip There are some problems arising due to the inclusion of the create button in smaller screens. When a user creates an exploration, the user can input data into the cards but the option to save changes is not shown. Even If the user edits the exploration, the changes cannot be saved due to lack of buttons on the navbar.
![screenshot from 2016-04-03 00 02 18](https://cloud.githubusercontent.com/assets/8845039/14228420/2906c4c6-f932-11e5-88a7-4f8fcbbbf30e.png)
 I think It is better to not give the option to edit/create explorations if the user in smaller screens. The smaller screen version will need a lot of scrolling to get a complete overview  of the exploration, which I think won't make a good user experience.

seanlip commented:
Yes, that's a good point. I had previously made the change with the idea that we'll be fixing the editor view later, anyway, but given that we want to do a release soon it seems sensible not to do this yet. Besides, it's quite straightforward to make the change in the future, so I've reverted it as you suggested. PTAL.

seanlip commented:
Based on previous comments, I'm going to regard this as LGTM and merge it, so that we can start preparing the release. (@giritheja, please let me know if this is incorrect, or if you'd like me to do anything further!)

seanlip commented:
Also, thanks for the review!

giritheja commented:
@seanlip Yeah. Anytime :+1:
